### PR TITLE
8290525: Move HeapRegion::_compaction_top to G1FullCollector

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this change that moves `HeapRegion::_compaction_top` into the `G1FullCollector` class; my reasoning is that these values are transient temporary values only ever useful for the full collection algorithm, basically results of the application of the algorithm, that are otherwise unused. This makes `HeapRegion` a bit easier to understand/read.

Testing: tier1-3

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8290525](https://bugs.openjdk.org/browse/JDK-8290525): Move HeapRegion::_compaction_top to G1FullCollector

### Reviewers
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**)
 * [Sangheon Kim](https://openjdk.org/census#sangheki) (@sangheon - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9560/head:pull/9560` \
`$ git checkout pull/9560`

Update a local copy of the PR: \
`$ git checkout pull/9560` \
`$ git pull https://git.openjdk.org/jdk pull/9560/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9560`

View PR using the GUI difftool: \
`$ git pr show -t 9560`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9560.diff">https://git.openjdk.org/jdk/pull/9560.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1FullCollector.hpp [hunk 1]
```diff
@@ -89,6 +90,8 @@ class G1FullCollector : StackObj {
 
   G1FullGCHeapRegionAttr _region_attr_table;
 
+  HeapWord*volatile *_compaction_tops;
```
Comment (kimbarrett) [inline src/hotspot/share/gc/g1/g1FullCollector.hpp:93]: Spacing in the type is strange.  `HeapWord* volatile* _compaction_tops;` is (I think) more usual for our code.

**Review by kimbarrett — APPROVED** [review]
Looks good.  Just one minor formatting nit.

**Review by kimbarrett — APPROVED** [review]

**Review by sangheon — APPROVED** [review]
Looks good.

**Comment by tschatzl** [comment]
Thanks @kimbarrett @sangheon for your reviews.

/integrate

**Pull request closed without merging**